packages/@webex/plugin-presence/src/presence.ts

```diff
@@ -0,0 +1,266 @@
+ */
+const Presence: IPresence = WebexPlugin.extend({
+  namespace: 'Presence',
+
+  children: {
+    batcher: PresenceBatcher,
+  },
+
+  session: {
+    worker: {
+      default() {
+        return new PresenceWorker();
+      },
+      type: 'any',
+    },
+  },
+
+  /**
+   * Initialize the presence worker for client
+   * @returns {undefined}
+   */
+  initialize(): void {
+    this.webex.once('ready', () => {
+      if (this.config.initializeWorker) {
+        this.worker.initialize(this.webex);
+      }
+    });
+  },
+
+  /**
+   * Trigger an event.
+   * @param {string} event
+   * @param {string} payload
+   * @returns {undefined}
+   */
+  emitEvent(event: string, payload: IEventPayload): void {
+    if (payload.type && payload.payload) {
+      this.trigger(event, payload);
+    }
+  },
+
+  /**
+   * Enables presence feature
+   * @returns {Promise<boolean>} resolves with true, if successful
+   */
+  enable(): Promise<boolean> {
+    return this.webex.internal.feature
+      .setFeature(USER, USER_PRESENCE_ENABLED, true)
+      .then((response) => response.value);
+  },
+
+  /**
+   * Disables presence feature
+   * @returns {Promise<boolean>} resolves with false, if successful
+   */
+  disable(): Promise<boolean> {
+    return this.webex.internal.feature
+      .setFeature(USER, USER_PRESENCE_ENABLED, false)
+      .then((response) => response.value);
+  },
+
+  /**
+   * Returns true if presence is enabled, false otherwise
+   * @returns {Promise<boolean>} resolves with true if presence is enabled
+   */
+  isEnabled(): Promise<boolean> {
+    return this.webex.internal.feature.getFeature(USER, USER_PRESENCE_ENABLED);
+  },
+
+  /**
+   * The status object
+   * @typedef {Object} PresenceStatusObject
+   * @property {string} url: Public resource identifier for presence
+   * @property {string} subject: User ID for the user the returned composed presence represents
+   * @property {string} status: Current composed presence state
+   * @property {string} statusTime: DateTime in RFC3339 format that the current status began
+   * @property {string} lastActive: DateTime in RFC3339 format that the service last saw activity from the user.
+   * @property {string} expires: DEPRECATED - DateTime in RFC3339 format that represents when the current
+   * status will expire. Will not exist if expiresTTL is -1.
+   * @property {Number} expiresTTL: TTL in seconds until the status will expire. If TTL is -1 the current
+   * status has no known expiration.
+   * @property {string} expiresTime: DateTime in RFC3339 format that the current status will expire. Missing
+   * field means no known expiration.
+   * @property {Object} vectorCounters: Used for packet ordering and tracking.
+   * @property {Boolean} suppressNotifications: Indicates if notification suppression is recommended for this status.
+   * @property {string} lastSeenDeviceUrl: Resource Identifier of the last device to post presence activity for
+   * this user.
+   */
```

Comment:
Should we consider converting this to a type or an interface?

Reply:
It is already converted and added.

Reply:
So, we are holding this just for documentation purposes? If so, I think we have a separate ticket to get a separate API reference like the Calling and EAF SDK for this.

Reply:
I kept them so that we don't lose anything. Maybe we can remove them when working on the API reference ticket.